### PR TITLE
Added react-router v4 with async load of components

### PR DESCRIPTION
**Pull request by chemitaxis**
Hi @Kikobeats I have added to the boilerplate react-router v4, I have added async load components for routing with needed (code splitting). I have added a NotMatch route too (https://reacttraining.com/react-router/web/example/no-match)

I would like to add in a future a more consistent router file.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "vsicons.presets.angular": false
```
Comment (Kikobeats) [inline .vscode/settings.json]: Can you remove this file from the PR, please? 😄

**Comment by chemitaxis** [comment]
Done @Kikobeats !! I will work this week i18n translations and a better code splitting with sass/less/css for production.

File: webpack.config.js [hunk 2]
```diff
@@ -26,7 +26,8 @@ module.exports = {
   ],
   output: {
     path: path.resolve('src/www'),
-    filename: 'assets/js/bundle.js'
```
Comment (Kikobeats) [inline webpack.config.js:29]: why renaming? I prefer a more predictable name

**Comment by chemitaxis** [comment]
In a future you could need two entry points, dont you think?

**Comment by Kikobeats** [comment]
make sense 👍

File: bin/server.js [hunk 3]
```diff
@@ -11,7 +11,18 @@ var server = new WebpackDevServer(compiler, {
   hot: true,
   historyApiFallback: true,
   overlay: true,
-  contentBase: 'src/www'
+  contentBase: 'src/www',
+  // If you want you can change the target URL for your backend API
+  // More information here:
+  // https://webpack.js.org/configuration/dev-server/#devserver-proxy
+  proxy: {
```
Comment (Kikobeats) [inline bin/server.js:18]: not sure about this change, can you explain why is necessary?

**Comment by Kikobeats** [comment]
Need to add `react-router-dom` as dependency!

**Comment by Kikobeats** [comment]
I need a little of time for understand better this PR.

Until this moment this project was feedback for another project:
https://github.com/windtoday/windtoday-marketplace

while I explore a feature, I add the feature in the boilerplate.

Definetly code splitting is 👌, but I'm a completly noob (because currently in my project I don't need it, I have just one view)

so sorry if I don't merge this quickly, I want to explore it 😄

**Comment by chemitaxis** [comment]
Of course!! I will try docuemnt in a better way, next week ;) dont worry!!

**Comment by Kikobeats** [comment]
Now with project wiht `jquery` is another project 😂

**Comment by chemitaxis** [comment]
This is just for testing ;) I don't know why this was a pull request!! hahahaha

**Pull request closed without merging**